### PR TITLE
fix: handle missing platform field in format_entity for unsupported entity types

### PR DESCRIPTION
**Pull request by alfiyas-datahub**
## Summary

Fixes `KeyError: 'platform'` crash in `format_entity()` when downstream lineage contains entity types not covered by the GraphQL query (e.g., DataJob, DataFlow).

## Problem

The `searchAcrossLineage` GraphQL query only includes `platform` field for specific entity types:
- `... on Dataset { platform { name } }`
- `... on Chart { platform { name } }`
- `... on Dashboard { platform { name } }`

When a **DataJob** (Airflow task) or **DataFlow** (Airflow DAG) appears in the downstream lineage, the response doesn't include the `platform` field, causing:

KeyError: 'platform'
File "rendering.py", line 22, in format_entity
platform = downstream["platform"]["name"]

## Solution

Added defensive null checking and URN-based fallback:

1. **Safe platform extraction**: Use `.get()` with fallback to URN parsing
2. **Safe name extraction**: Use `.get()` with fallback to URN
3. **Safe type extraction**: Added null checks for `subTypes.typeNames`
4. **New helper**: `_extract_platform_from_urn()` extracts platform info from URN string
5. **Consistent URN access**: Extract URN once at start and use consistently throughout
6. **Correct orchestrator detection**: Parse URN structure to extract orchestrator from correct position (not from DAG names)

---

> [!NOTE]
> **Medium Risk**
> Touches rendering/formatting used for lineage display and introduces URN parsing + logging; behavior changes could affect output formatting across entity types, but is well covered by new tests.
> 
> **Overview**
> Prevents `format_entity()` from crashing when downstream lineage entities omit `platform`/`properties` (e.g., `DATA_JOB`, `DATA_FLOW`) by switching to defensive `.get()` access, requiring `urn`, and falling back to using the URN as the display name.
> 
> Adds `_extract_platform_from_urn()` (with warning logs) to derive platform/orchestrator from DataHub URNs when `platform` is missing, improves type formatting (`DATA_JOB` -> `Data Job`) and `subTypes` handling, and expands tests substantially to cover these edge cases plus URL suffix behavior.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 3469a77c2821c8cd1faacbf4cb23c6c7e4478680. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/rendering.py [hunk 1]
```diff
@@ -19,15 +22,97 @@ def datahub_url_from_urn(
...
+                value = getattr(urn_obj, attr, None)
+                if value:
+                    # If it's a platform URN, extract the platform name
+                    if "dataPlatform:" in str(value):
+                        platform_urn = Urn.from_string(str(value))
+                        return platform_urn.entity_ids[0].capitalize()
+                    return str(value).capitalize()
+            except (AttributeError, IndexError):
+                continue
+
```
Comment (sgomezvillamor) [inline src/rendering.py:116]: Thinking on how to make this resilient to future entities, should we add a final fallback value or maybe let it fail? 
I mean, what if `orchestrator`, `platform` or `dashboard_too` do not apply neither? what should happen in that case?

**Review by sgomezvillamor — APPROVED** [review]
LGTM

File: src/rendering.py (hunk 1 above)
Comment (alfiyas-datahub) [reply src/rendering.py:116]: Currently, it falls through and returns "Unknown" silently. I'll add a warning log when we encounter an entity type we don't handle, so we have visibility without breaking the user's CI/CD.
